Packs/ZoomMail/Integrations/ZoomMail/ZoomMail.py

```diff
@@ -0,0 +1,1349 @@
+
+    now_epoch = datetime.now().timestamp()
+    now_utc = convert_to_utc(now_epoch)
+
+    if not last_fetch_info["internalDate"]:
+        first_fetch_dt = parse(first_fetch_time)
+        if not first_fetch_dt:
+            demisto.info("First fetch time is empty or failed to be parsed, defaulting to 3 days.")
+            first_fetch_dt = datetime.now() - timedelta(days=3)
+        last_fetch_info["internalDate"] = first_fetch_dt.timestamp()
+
+    incidents: list[dict[str, Any]] = []
+
+    last_fetch_time_utc = convert_to_utc(last_fetch_info["internalDate"])
+    query = f"{fetch_query} after:{last_fetch_time_utc} before:{now_utc}"
+
+    if last_page_fetch_token:
+        demisto.info(f"Fetching additional messages from previous fetch run, {last_page_fetch_token}")
+        messages_response = client.list_emails(
+            email=fetch_from,
+            page_token=last_page_fetch_token,
+            label_ids=fetch_labels,
+            max_results=str(max_fetch),
+        )
+    else:
+        messages_response = client.list_emails(
+            email=fetch_from,
+            max_results=str(max_fetch),
+            query=query,
+            label_ids=fetch_labels,
+        )
+    next_page_token = messages_response.get("nextPageToken", None)
+    messages = messages_response.get("messages", [])
+    message_dates: list[float] = []
+
+    demisto.info(f"Found {len(messages)} messages to process prior to filtering.")
+
+    for msg in messages:
+        message_id = msg.get("id")
+        thread_id = msg.get("threadId", "")
+        message_details = client.get_email_message(
+            email=fetch_from, message_id=message_id
+        )
+        internal_date = float(message_details.get("internalDate")) / 1000.0
+
+        msg.update({"internalDate": internal_date})
+
+        # Check if it's a new message, and if it's either all messages (fetch_threads is False) or only thread starters
+        if (internal_date > last_fetch_info["internalDate"] or (
+            internal_date == last_fetch_info["internalDate"] and message_id not in last_fetch_info["ids"]
+        )) and (fetch_threads or (not fetch_threads and message_id == thread_id)):
+            incident = zoom_mail_to_incident(message_details, client, fetch_from)
+            incidents.append(incident)
+            if not last_page_fetch_token:
+                message_dates.append(internal_date)
+
+    if message_dates and not next_page_token:
+        new_internal_date = max(message_dates)
+        new_ids = [
+            msg["id"] for msg in messages if msg["internalDate"] == new_internal_date
+        ]
+
+        last_fetch_info = {"internalDate": new_internal_date, "ids": new_ids}
+
+    # If we don't collect any new messages, but we have a next fetch token,
+    # we are fetching old messages and need to stop.
+    if len(incidents) == 0 and next_page_token:
+        demisto.info("No new messages found, stopping fetch.")
+        next_page_token = None
```

Comment:
since time filtering works, we should go over all pages when they exist. and not to delete next_page_token. also I do not understand why it is relevant that no incident were found to delete the next page.

Reply:
on second glance you just need to explain here that pages ae arranged backwards by time, so if all messages has been filtered, new new messages will arrive in the next page because thy are older.